zad2/warriors.rb

```diff
@@ -1,12 +1,75 @@
-class Warrior
+class Character
+  attr_accessor :name, :level
+
   def initialize(name:, level:)
+    @name = name
+    @level = validate_level(level) ? level : 1
+  end
+
+  def card
+    "#{@name} (lvl #{@level})"
   end
 
   def strength
+    [*1..12].sample + @level
+  end
+
+  private
+
+  def validate_level(level)
+    level.between?(1, 99)
+  rescue
+    false
   end
 end
 
+class Warrior < Character
+  def level_up(n)
+    @level = @level + n <= 99 ? @level + n : 99
```

Comment:
to już może być trochę za dużo jak na wyrażenia tego typu, one powinny być bardzo krótkie, aby zachować czytelność. Dobrym przykład jest `Character#initialize`
`validate_level(level) ? level : 1`